### PR TITLE
Fixed version number error in setup.py and fixed CI on travis

### PR DESCRIPTION
**Pull request by Bounder**
# Description

These changes fix a non existing version in setup.py, which causes problems when installing this package using pipenv. Also CI building has been fixed and the project is now tested against 1.8 and 1.11

# Motivation

The original intent was fixing a not existing version number in setup.py which causes problems when installing this project using pipenv. However when forking the project I saw the CI was broken so I tried to fix that. I'm using this package in a lot of my project so it seemed logical to get this fixed and make sure it works in the python versions that are currently mainstream.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,39 @@
 language: python
 
+python:
```
Comment (Bouke) [inline .travis.yml]: Adding this will create additional entries in the build matrix. See also [the ci build](https://travis-ci.org/Bouke/django-airbrake/builds/350670567?utm_source=github_status&utm_medium=notification). No need to add this list of python versions, and control the build matrix using the `matrix:` field.

File: setup.cfg [hunk 2]
```diff
@@ -1,2 +1,2 @@
 [flake8]
-max-line-length = 100
+max-line-length = 119
```
Comment (Bouke) [inline setup.cfg]: Make Guido proud by staying with 100 👍 .

File: setup.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
     description='Airbrake exception logger for Django',
     license='MIT',
     long_description=open('README.rst').read(),
-    install_requires=['Django>=1.8,<1.9.99'],
+    install_requires=['Django>=1.8,<=1.11.11'],
```
Comment (Bouke) [inline setup.py]: I used 1.9.99 to specify all version up to/excluding 1.10 (and pre-releases of 1.10). Also specifying 1.11.11 is problematic when a bugfix is released (1.11.12) which would be excluded.

As 1.8 has almost reached EOL (April 1st), we can specify 1.11 as the lower bound and not have an upper bound.

File: tests/tests.py [hunk 4]
```diff
@@ -17,24 +17,7 @@
...
-            raise Exception('Could not find my shoes')
-        except Exception as exc:
-            self.logger.exception(exc)
-
-        self.assertEqual(urlopen.call_count, 1)
-        xml = urlopen.call_args[0][0].data
-        xsd_validate(xml)
-
-        self.assertTrue(xml_compare(etree.fromstring("""
+DJANGO_18_RESPONSE = """
```
Comment (Bouke) [inline tests/tests.py]: As we're dropping support for 1.8, is this still needed?

**Review by Bouke — CHANGES_REQUESTED** [review]
Hey man, mooie PR! 💯 Ik kwam een paar dingetjes tegen bij de review.

- Can we also have a test for Django 2.0? If the tests fail, we can mark them as expected failure in Travis, so CI doesn't mark the build as failed.
- Can you update the readme to reflect the updated list of supported Django / Python versions?

**Comment by Bounder** [comment]
I added the requested changes except for the max-line-length as discussed. Project is now Django>=1.11,<=2.0.5 compatible. Dropped support for 1.8 as the LTS was dropped this month

File: Pipfile.lock [hunk 5]
```diff
@@ -0,0 +1,206 @@
+{
```
Comment (Bouke) [inline Pipfile.lock]: It's not common practice for libraries to ship a lockfile. Lockfiles are great for end-users who want to have a reproducible build, with specific versions of dependencies. In this case, we should not include the lockfile.

**Comment by Bounder** [comment]
Removed piffle.lock from git repo as requested

File: setup.py [hunk 6]
```diff
@@ -13,7 +13,7 @@
     description='Airbrake exception logger for Django',
     license='MIT',
     long_description=open('README.rst').read(),
-    install_requires=['Django>=1.8,<1.9.99'],
+    install_requires=['Django>=1.11,<=2.0.5'],
```
Comment (Bouke) [inline setup.py]: Requiring up to 2.0.5 will cause problems once 2.0.6 or 2.1 is released. Probably better to not specify an upper bound.

File: tests/utils.py [hunk 7]
```diff
@@ -37,11 +37,13 @@ def xml_compare(x1, x2, reporter=None):
...
         for c2 in cl2:
-            if xml_compare(c1, c2, reporter=reporter):  # no reporter, fail silently
-                cl2.remove(c2)
+            if xml_compare(c1, c2):  # no reporter, fail silently
+                # cl2.remove(c2)
                 break
         else:
             if reporter:
+                xml_compare(c1, x2.getchildren()[i], reporter=reporter)
+                # xml_compare(c1, cl2[0], reporter=reporter)
```
Comment (Bouke) [inline tests/utils.py]: Why are these changes needed?

File: .travis.yml [hunk 8]
```diff
@@ -3,24 +3,29 @@ language: python
...
-      env: DJANGO=1.9 COVERAGE=YES
+      env: DJANGO=2.0
+    - python: "3.6"
+      env: DJANGO=2.0 COVERAGE=YES FLAKE8=YES
 
-    - python: "3.4"
-      env: FLAKE8=YES
+before_install:
+  - sudo apt-get -qq update
+  - sudo apt-get install -y libxml2-utils
```
Comment (Bouke) [inline .travis.yml]: Does this PR now depend on `libxml2-utils` being available? Or only for testing?

File: setup.cfg (hunk 2 above)
Comment (Bouke) [reply setup.cfg]: Can you revert this? Or does it introduce errors for existing code?

**Review by Bouke — CHANGES_REQUESTED** [review]
Great PR so far. A few minor changes and I'll merge it in!

File: .travis.yml (hunk 8 above)
Comment (Bounder) [reply .travis.yml]: This is needed since you changed the way to validate the xml, see commit: 69dc92c5cb557a7bd09b9a9316d93d76c6d2e390

File: tests/utils.py (hunk 7 above)
Comment (Bounder) [reply tests/utils.py]: Don't know anymore, this was however how it originally was, so probably changed it to make the test run again. Maybe you can tell me why it was changed?

File: .travis.yml (hunk 8 above)
Comment (Bounder) [reply .travis.yml]: So yes, only for testing. It's not used by airbrake itself.

**Comment by Bounder** [comment]
Thanks for being such a PITA, reviewing this PR. This is the kind of code quality control and peer reviewing I'm missing in my day-to-day programming. Learned a lot again! Thanks.

**Review by Bouke — APPROVED** [review]

**Comment by Bouke** [comment]
I removed `Pipfile` from the repo. Would a library want to provide something like that? It seems to duplicate `requirements_dev.txt` (probably a good thing, right?), but also `setup.py` (it was still listing `django<=2.0.5`). I might need to familiarize myself with 2018's Python packaging status-quo.
